firebase-config/src/test/java/com/google/firebase/remoteconfig/FirebaseRemoteConfigTest.java

```diff
@@ -1039,6 +1043,45 @@ public void personalization_hasMetadata_successful_without_analytics() throws Ex
             });
   }
 
+  @Test
+  public void realtime_addListener_success() {
+    ConfigUpdateListener eventListener = generateEmptyRealtimeListener();
+    when(mockRealtimeHttpClient.addRealtimeConfigUpdateListener(eventListener))
+        .thenReturn(
+            new ConfigUpdateListenerRegistration() {
+              @Override
+              public void remove() {}
+            });
+    ConfigUpdateListenerRegistration registration =
+        mockRealtimeHttpClient.addRealtimeConfigUpdateListener(eventListener);
+    verify(mockRealtimeHttpClient).addRealtimeConfigUpdateListener(eventListener);
+    assertThat(registration).isNotNull();
+  }
+
+  @Test
+  public void realtime_addListener_fail() {
+    when(mockRealtimeHttpClient.addRealtimeConfigUpdateListener(null)).thenReturn(null);
+
+    ConfigUpdateListenerRegistration registration =
+        mockRealtimeHttpClient.addRealtimeConfigUpdateListener(null);
+
+    assertThat(registration).isNull();
+  }
+
+  @Test
+  public void realtime_removeListener_success() {
+    ConfigUpdateListener eventListener = generateEmptyRealtimeListener();
+    when(mockRealtimeHttpClient.addRealtimeConfigUpdateListener(eventListener))
+        .thenReturn(mockRealtimeRegistration);
+
+    ConfigUpdateListenerRegistration registration =
+        mockRealtimeHttpClient.addRealtimeConfigUpdateListener(eventListener);
+    registration.remove();
+
+    assertThat(registration).isNotNull();
```

Comment:
I think this is redundant since otherwise line 1079 would have thrown NullPointerException